pkg/server/batch_check.go

```diff
@@ -0,0 +1,130 @@
+package server
+
+import (
+	"context"
+	"errors"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/openfga/openfga/pkg/middleware/validator"
+
+	"github.com/openfga/openfga/internal/condition"
+	"github.com/openfga/openfga/internal/graph"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+
+	"github.com/openfga/openfga/internal/authz"
+	"github.com/openfga/openfga/pkg/server/commands"
+	"github.com/openfga/openfga/pkg/telemetry"
+)
+
+func (s *Server) BatchCheck(ctx context.Context, req *openfgav1.BatchCheckRequest) (*openfgav1.BatchCheckResponse, error) {
```

Comment:
Yeah I separated it mostly for ease of use. Imo every endpoint should probably live in its own file inside `/server`, it's kind of the opposite of cramming them all into one. ListUsers probably got missed because it was the one exception to the pattern, if **every** endpoint was its own file they would probably all get covered in a case like that.

All that said, that rearranging is definitely not part of this PR. I wouldn't mind doing that refactor work as soon as batch check lands tho 🤔

Reply:
Sounds good. Let's discuss as a team what we prefer and then we can make changes, if any.

Reply:
> Can we consider putting this on server.go like the rest? (cc @willvedd WDYT? i think we should eventually move ListUsers to server.go as well)

I'm not convinced that placing all code in the same file is a legitimate way of combatting bugs 🫠 . That single file is so long that I'm actually of the opinion that it creates more bugs.